modelexpress_client/python/modelexpress/nixl_transfer.py

```
@@ -271,6 +292,61 @@ def _find_contiguous_regions(
+        size_mismatches: list[tuple[int, int, int]] = []
+        for i, (src_region, (_local_addr, local_size)) in enumerate(
+            zip(source_regions, local_regions, strict=True)
+        ):
+            if src_region.size != local_size:
+                size_mismatches.append((i, src_region.size, local_size))
+
+        if size_mismatches:
+            # Log just the first few so errors stay readable.
+            head = size_mismatches[:5]
+            sample = ", ".join(
+                f"region {i}: source={s} local={l}" for i, s, l in head
```

Ruff (0.15.10)

[error] Ambiguous variable name: `l`

(E741)